klog_test.go

```diff
@@ -953,6 +953,63 @@ func TestKRef(t *testing.T) {
 	}
 }
 
+func TestResourceRefString(t *testing.T) {
+	tests := []struct {
+		testname  string
+		group     string
+		version   string
+		kind      string
+		name      string
+		namespace string
+		want      string
+	}{
+		{
+			testname:  "with group, version, kind, namespace and name",
+			group:     "test-group",
+			version:   "test-version",
```

Comment:
Might be nitpicking here but it would be really nice to use the GVK values that are more k8s styled. 

> Versions start with a v followed by a number, an optional beta or alpha designation, and optional additional numeric versioning information

xref: https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#version-priority